### ISSUE TITLE
Accessibility issue: Button contrast is too low

### ISSUE THREAD

**Issue by nh916** [materializecss/materialize#103]
The default button background color and foreground color has too low of a contrast that is a problem and makes our websites inaccessible for people with any kind of blindness or color deficiency. We are currently failing all accessibility color contrast checks.

The background color of: _#26A69A_
and foreground color of: _#FFFFFF_

results in a contrast ratio of: _**2.99:1**_

We want the contrast ratio to be something around **7**

We can achieve this contrast ratio by changing up the darkness of the background to something like

The background color of: _#0F614A_
and foreground color of: _#FFFFFF_

results in a contrast ratio of: _**7.42:1**_

Another great way of fixing this problem that is easier is to make the text black by default on buttons, instead of the current default white text color on buttons, and keep the same default button color and that would get us exactly _**7**_ contrast ratio. 

The more contrast ratio the better but I think _**7**_ gets us the minimum we need to get what we want.

Here is the resources/contrast checker that I am using to try to make our website more accessible.
https://webaim.org/resources/contrastchecker/

Here is our current button contrast ratios. 
https://imgur.com/a/1ltbprV 
![image](https://user-images.githubusercontent.com/26590757/116593579-155fce00-a8d6-11eb-86fe-604d6bc06d59.png)

**Comment by nh916** [comment]
I checked around at the other buttons as well, and after checking other button defaults like the _**floating button**_ and the _**disabled buttons**_ 

### Disabled Buttons Contrast Ratio
https://imgur.com/a/anRvtxO

### Floating Buttons Contrast Ratio
https://imgur.com/a/Ectr4YW

**Comment by ChildishGiant** [comment]
Whilst having the library be highly accessible would be amazing, I fear that we could get too caught up in numbers. a 7:1 ratio is what you need for full coverage of the WCAG but if we cater to all of the specifications (some of which are designed for those with the most severe impairments, whilst using the smallest of text), we'd end up with a site that only uses black and white. The disabled buttons are supposed to have low contrast as they are not usable, although I do agree that maybe it needs to be more legible.

**Comment by nh916** [comment]
I do agree that we shouldn't worry too much about the numbers, and there are going to be some places where it wont be accessible and we'll just have to leave it as is, but I think the more accessible that we make it the better if we can. If we cannot possibly do so and we tried, then I will happily leave it and say that we tried, but we just couldn't make it happen, and thats that.
I think it could be fun getting creative with how to balance the UI so it looks beautiful to a abled person and a disabled person can easily work with it too. 

I think the future of web will be one day where people who are abled can use the internet without a problem and when they become disabled they can use the web differently but have the same experience and never feel like they are losing out on anything.

Also, a side effect of have good accessibility is that it tends to create better UI and UX by just implementing good accessibility practices

**Comment by DanielRuf** [comment]
I think it makes sense to reconsider a high contrast version for the future. But at the moment it could introduce new issues. We will have to check how Google Material defines the colors.

**Comment by wuda-io** [comment]
Check out the new colors! Definitely better contrast ratios!
Also Dark Mode is now available. Maybe you find some ratios which are still not matching. Glad for every input.
Please re-open if i missed something. Thanks for the effort!
